src/base/static/js/views/geometry-editor-view.js

```diff
@@ -0,0 +1,583 @@
+/* A view enabling the creation and editing of geometry layers */
+
+module.exports = Backbone.View.extend({
+  events: {
+    "click .create-marker": "onClickCreateMarker",
+    "click .create-polyline": "onClickCreatePolyline",
+    "click .create-polygon": "onClickCreatePolygon",
+    "click .edit-geometry": "onClickEditGeometry",
+    "click .delete-geometry": "onClickDeleteGeometry",
+    "click .colorpicker": "onClickColorpicker",
+    "change input[name='geometry']": "onIconSelection"
+  },
+  initialize: function() {
+    var self = this;
+
+    this.map = this.options.map;
+    this.workingGeometry = null;
+    this.numVertices = 0;
+    this.layerType = null;
+    this.iconUrl = null;
+    this.existingLayerView = null;
+    this.isEditing = false;
+    this.editingLayerGroup = new L.FeatureGroup();
+    this.DRAWING_DEFAULTS = {
+      color: "#f06eaa",
+      opacity: 0.5,
+      fillColor: "#f06eaa",
+      fillOpacity: 0.2,
+      fill: {
+        color: "fillColor",
+        opacity: "fillOpacity"
+      },
+      stroke: {
+        color: "color",
+        opacity: "opacity"
+      }
+    };
+    this.colorpickerSettings = {
+      editMode: "fill",
+      color: this.DRAWING_DEFAULTS.color,
+      opacity: this.DRAWING_DEFAULTS.opacity,
+      fillColor: this.DRAWING_DEFAULTS.fillColor,
+      fillOpacity: this.DRAWING_DEFAULTS.fillOpacity
+    };
+
+    // Init the colorpicker
+    $("body").append("<div class='toolbar-colorpicker-container'></div>");
```

Comment:
I guess there's not really a good reason it's not in a template. It's so tiny that I thought a whole template might be overkill, but I agree it'd be more consistent to make a template here. Resolved.